Library/OcCpuLib/OcCpuLib.c

```diff
@@ -884,27 +880,33 @@ ScanAmdProcessor (
         //
         if (Cpu->MaxExtId >= 0x8000001E) {
           AsmCpuid (0x8000001E, NULL, &CpuidEbx, NULL, NULL);
-          Cpu->CoreCount =
-            (UINT16) DivU64x32 (
-              Cpu->ThreadCount,
-              (BitFieldRead32 (CpuidEbx, 8, 15) + 1)
-            );
+          Cpu->CoreCount = (UINT16)DivU64x32 (Cpu->ThreadCount, (BitFieldRead32(CpuidEbx, 8, 15) + 1));
         }
         break;
-      case 0x06:
+      case 0x06: // 15h_16h Family
       case 0x07:
         CofVid           = AsmReadMsr64 (K10_COFVID_STATUS);
         CoreFrequencyID  = BitFieldRead64 (CofVid, 0, 5);
         CoreDivisorID    = CofVid & BIT6;
         Divisor          = LShiftU64 (1, CoreDivisorID);
-        Cpu->MaxBusRatio = (UINT8)(((CoreFrequencyID + 16) / Divisor) + 1);
+        Cpu->MaxBusRatio = (UINT8) ((CoreFrequencyID + 16) / (2 * Divisor));
```

Comment:
This change does not make sense to me, as well as the original code. Check [AMD BIOS and Kernel developer guide](http://support.amd.com/TechDocs/42300_15h_Mod_10h-1Fh_BKDG.pdf). On pages 582 and 583 `MSRC001_0071 COFVID Status` description is given. Divisor is not just bit 6, but it actually is three bits from 6:8. Yet even if we correct that, it looks like neither value is directly related to maximum bus multiplier, which is `MaxCpuCof` for newer models, i.e. bits from 49:54.

Maybe I am wrong, but I believe the code above is meant to be written as follows:
```C
CofVid            = AsmReadMsr64 (K10_COFVID_STATUS);
Cpu->MaxBusRatio  = BitFieldRead64 (CofVid, 49, 54);
```